flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/FileMergingSnapshotManagerBase.java

```diff
@@ -569,6 +581,67 @@ public void reusePreviousStateHandle(
         }
     }
 
+    // ------------------------------------------------------------------------
+    //  Space Control
+    // ------------------------------------------------------------------------
+
+    /**
+     * The core method that control space if needed. This method will compare the desired space
+     * amplification with current one, and if it exceeds the configured amplification, this method
+     * will mark minimal set of {@link PhysicalFile}s not to be reused anymore.
+     */
+    private void controlSpace() {
+        if (spaceStat.logicalFileSize.get() * maxSpaceAmplification
```

Comment:
Since there still be another code getting value from `spaceStat` and calculate with `maxSpaceAmplification` (Line 598), I'd suggest keep this as it is. Could you elaborate more on the metrics?

Reply:
Okay, I think we could consider it later in the metric PR.